lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -177,6 +189,192 @@ impl EnsoLexer {
 }
 
 
+// === Comments ===
+
+/// The set of rules for lexing Enso comments.
+#[allow(dead_code)]
+impl EnsoLexer {
+
+    // === Disable Comments ===
+
+    /// Triggered when a disable comment is encountered.
+    fn on_begin_disable_comment<R:LazyReader>(&mut self, _reader:&mut R){
+        trace!(self.logger,"Comment::on_begin_disable_comment");
+        let offset          = self.offset.consume();
+        let disable_comment = self.disable_comment;
+        self.comment_state.set_offset(offset);
+        self.push_state(disable_comment);
+    }
+
+    /// Accumulate the disable comment contents.
+    fn on_build_disable_comment<R:LazyReader>(&mut self, _reader:&mut R){
+        trace!(self.logger,"Comment::on_build_disable_comment");
+        let current = self.consume_current();
+        self.comment_state.append_to_line(current);
+    }
+
+    /// Triggered when a disable comment is ended.
+    fn on_end_disable_comment<R:LazyReader>(&mut self, line_end:token::LineEnding, reader:&mut R) {
+        trace!(self.logger,"Comment::on_end_disable_comment");
+        self.discard_current();
+        let disable_comment = self.disable_comment;
+        let comment         = self.comment_state.consume_current();
+        let token           = comment.into();
+        self.append_token(token);
+        self.pop_states_including(disable_comment);
+        match line_end {
+            token::LineEnding::None => self.on_eof(reader),
+            _                       => {
+                let block_seen_newline = self.block_newline;
+                self.block_state.push_line_ending(line_end);
+                self.block_submit_line(reader);
+                self.push_state(block_seen_newline);
+            },
+        }
+    }
+
+    // === Doc Comments ===
```

Comment:
Here and in other files / places - these sections have 2 newlines above, big sections have 3